### PR TITLE
Use ':' as separator for name:stream:version in pungi variants.xml when mashing modules.

### PR DESCRIPTION
**Pull request by jankaluza**
Signed-off-by: Jan Kaluza <jkaluza@redhat.com>

### CODE REVIEW

**Comment by puiterwijk** [comment]
@centos-ci add to whitelist

File: bodhi/server/consumers/masher.py [hunk 1]
```diff
@@ -1289,10 +1320,18 @@ def _generate_module_list(self):
...
+            for update in self.compose.updates:
+                for build in update.builds:
+                    koji.getBuild(build.nvr)
+            results = koji.multiCall()
+            for result, build in zip(results, update.builds):
+                self._raise_on_get_build_multicall_error(result, build)
+                koji_build = result[0]
+                # name:stream:version maps to Koji's name-version-release.
+                ns = "%s:%s" % (koji_build["name"], koji_build["version"])
+                version = koji_build["release"]
```
Comment (bowlofeggs) [inline bodhi/server/consumers/masher.py]: This code block is almost identical to the code block above. Let's replace it with a method so we can adjust them together in the future if needed.

**Review by bowlofeggs — CHANGES_REQUESTED** [review]
There are tests that assert that exceptions are raised by ```_raise_on_get_build_multicall_error```, but we will also need tests that assert that the correct data is returned (with :'s) from ```_generate_module_list``` when there are not koji errors.

File: bodhi/server/consumers/masher.py [hunk 2]
```diff
@@ -1284,12 +1289,43 @@ def _raise_on_get_build_multicall_error(self, result, build):
             self.log.error(err)
             raise Exception(err)
 
+    def _add_build_to_newest_builds(self, newest_builds, koji_build, override=False):
+        """
+        Add Koji build to newest_builds dict if it's newer than the one there or override is set.
+
+        Args:
+            newest_builds (dict): Dict with name:stream as a key and version as a value.
```
Comment (bowlofeggs) [inline bodhi/server/consumers/masher.py]: It looks like the value is actually a dictionary with the key "version" (perhaps among others?) based on reading the code below:

```
curr_version = newest_builds[ns]['version']
```

**Review by bowlofeggs — COMMENTED** [review]
Just a nitpick on a docblock.

I'm running the tests now on my laptop.

**Review by bowlofeggs — APPROVED** [review]
Alright, I've run CI for this on my laptop and it passes, so I think we are good to merge. Please correct that one docblock I mentioned earlier, but no need for re-review.

**Pull request merged**